### DISCUSSION TITLE
Proper Handling of Repeating Triggers on the 31st of Every Month

### DISCUSSION THREAD

**Discussion by gjdownes** [quartznet/quartznet#1053] [Q&A] [answered]
Is there a best practice for setting up a monthly job that executes on or after the 29th of every month, e.g. `0 0 31 */1 *`? 

There seem to be limitations with CRON to make this work.

Since RRULE has support for these types of scenarios, I was considering using an RRULE parser (e.g. iCal.net) to generate the next upcoming recurrence, and continue scheduling individually one after the other is triggered.

Any tips would be appreciated, using CRON or otherwise.

**Comment by lahma** [comment]
Sorry I don't have an answer off the shelf, but I'm curious of this business rule. Why run this only on every four years in February?

**Reply by gjdownes** [reply]
Hi Marko,

The expectation is the events will occur every month on a given day (e.g. 29th), so Jan 29, Feb 28/29, Mar 29, etc. I'm only highlighting the 29th because it's a worst case scenario.

**Comment by lahma** [comment, marked as answer]
So if you are looking for last day of month (run once in month), it would probably be for 3am:  `0 0 3 L * * *`
